api/src/main/java/org/apache/iceberg/DeleteFiles.java

```diff
@@ -55,6 +55,17 @@ default DeleteFiles deleteFile(DataFile file) {
     return this;
   }
 
+  /**
+   * Delete a file tracked by a {@link DeleteFile} from the underlying table.
+   *
+   * @param file a DeleteFile to remove from the table
+   * @return this for method chaining
+   */
+  default DeleteFiles deleteFile(DeleteFile file) {
+    deleteFile(file.path());
```

Comment:
This will not work, and needs to be changed to use the delete file specific API. and I'm surprised that UT didn't catch it.